### ISSUE TITLE
Add an AUR package

### ISSUE THREAD

**Issue by powwu** [Drewol/unnamed-sdvx-clone#194]
Consider adding a package to the Arch User Repository, or AUR, to assist users with easily installing your package from source. The AUR has over 53k packages, and adding yours would make it just that much better. More information can be found [here](https://aur.archlinux.org/).

**Comment by mirrornoir** [comment]
I've been thinking of making an AUR package for a while but never got around to it. Main barrier is I wasn't sure where it should go in the file system, but `~/.local/share/unnamed-sdvx-clone` seems pretty sane. I'll try making a package today.

It should be said that it's pretty rare that upstream also maintains the AUR package, it's usually up to the users themselves.

**Comment by Drewol** [comment]
I'm planning to soon to over how things are linked and maybe change some things around so some things that are dynamically linked might become statically linked and I'm going to see if I can reduce the things we build so some dependencies might change soon

**Comment by mirrornoir** [comment]
So if I understand correctly, moving things from git submodules into normal dependencies? I'd still like to get an AUR package out but that kind of change should be trivial.

**Comment by Drewol** [comment]
yeah pretty much, and something might become statically linked so that you don't have to worry about the library version

**Comment by mirrornoir** [comment]
So I did create an AUR package but it was a real ugly hack. USC currently isn't very friendly to filesystem concepts and I'm not really willing to release an AUR package until that situation changes.

My recommendation to you @powwu for now is just to continue using git to manage this project, then you're free to copy the `bin` directory to wherever you see fit. Keep in mind you can do things like `git pull` to update your local copy. Until things change, git is still the best method to update and manage this project.

**Comment by Drewol** [comment]
I've made some changes locally that makes it so the game so it doesn't require your cwd to be the game's bin folder anymore, I'm going to do some more testing before I push that, but what more needs to be done before a decent package can be made?

**Comment by mirrornoir** [comment]
Awesome! Yeah, that is a big help. We can take this further using proper config directories utilizing concepts like `XDG_CONFIG_HOME` in Linux and `%APPDATA%` in Windows which would be great for packaging and being more friendly to file system concepts. 

An ideal setup could be achieved on Linux for example by the game data like fonts and skins living in the system location `/usr/share/unnamed-sdvx-clone` and the game its self located in `/usr/bin/usc-game`. Then when run, USC can check if `XDG_CONFIG_HOME/unnamed-sdvx-clone` exists and create one for user settings if it doesn't. The user could then make their own `skins` and `fonts` folder that would supplement the default system folders. Typically apps with that kind of setup use cmake install scripts to install to these locations, directories configurable in case other Linux distributions have different file system and packaging concepts. This kind of setup would make an AUR package trivial and friendly to concepts and style guides.

A similar setup could be achieved in Windows if the game is installed to non user writable directories like `C:\Program Files` and then creating `%APPDATA%\unnamed-sdvx-clone` if it doesn't exist.

Another point I'd like to make is that dependencies are best handled on Linux systems by using shared system libraries. Not every dependency used by USC is available on Linux as packages but things like `libarchive` and `discord-rpc` are readily available. I'm not a developer and I have no idea how this would be handled cross-platform but it would be great if shared libraries could be used if available. In the context of this AUR package I could list `libarchive` and `discord-rpc` as a dependency for example.

I know these are some non-trivial changes but I'm excited to see USC play nice with operating system file system concepts. I hope these suggestions help!

**Comment by Drewol** [comment]
Someone made an AUR package after the addition of `-gamedir` but it would still be good if that wasn't necessary https://aur.archlinux.org/packages/unnamed-sdvx-clone/

**Comment by scatterfish** [comment]
Co-maintainer of that package here. Currently we don't use `-gamedir` and instead just install to `/opt/`. Correct me if I'm wrong, but the addition of `-gamedir` doesn't do much at all because what we really need is for the game to grab resources from various places, instead of just a single directory.

**Comment by Drewol** [comment]
the game also needs write permissions for config, log, db, and screenshots, moving only those could be an option. The songs path is already configurable, then I think it's mostly just skin installations that can be tricky.

**Comment by scatterfish** [comment]
It's a terrible hack, but our current solution to get everything working is to just grant those write permissions despite `/opt/` being a protected place. I agree that `/usr/share/` is the best place for application data, but that's still a protected location so songs and skins couldn't go there even though it seems like they belong there.

`~/.local/share/` is a reasonable alternative, but the game itself will need to look for it or we'll need to put a wrapper script instead of the game itself in `/usr/bin/`.

A bigger issue however is that of untracked files. Maybe moving to a better directory will fix it, but since USC has a payload, anything written to package directories after installation isn't tracked by `pacman` and can't be removed by `pacman`. Currently unsure what to do about it.
